### PR TITLE
refactor(navigation): Mon 12052 global left menu

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
global left menu rework

### CODE REVIEW

**Comment by Noha-ElAbrouki** [comment]
here are some screens : 
dark mode : 
regular collaps
regularCollaps2

collapsWithSubheaderDark

light mode : 
regularCollapsLight

regularCollaps2Light
collqpsWithSubHeaderLight

File: www/front_src/src/Navigation/Sidebar/Menu/ListButton.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import classnames from 'classnames';
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/ListButton.tsx:3]: Use clsx instead

File: www/front_src/src/Navigation/Sidebar/Menu/ListButton.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    alignItems: 'center',
+  },
+  icon: {
+    color: theme.palette.text.primary,
+  },
+}));
+
+const ListButton = ({
+  onMouseEnter,
+  handlClickItem,
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/ListButton.tsx:50]:
```suggestion
  onClick,
```

File: www/front_src/src/Navigation/Sidebar/Menu/MinCollaps.tsx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+      width: theme.spacing(1.2),
+    },
+
+    '&::-webkit-scrollbar-thumb': {
+      backgroundColor: theme.palette.action.disabled,
+      borderRadius: 5,
+    },
+    '&::-webkit-scrollbar-track': {
+      borderRadius: 5,
+      boxShadow: `inset ${theme.spacing(0.5, 0.5, 0.5, 0.5)} ${
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/MinCollaps.tsx:84]: We should use material-ui shadow mechanism instead

File: www/front_src/src/Navigation/Sidebar/Menu/MinCollaps.tsx [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    position: 'fixed',
+    top: ({ currentTop }: StyleProps): number | undefined => currentTop,
+    width: theme.spacing(25.5),
+    zIndex: theme.zIndex.mobileStepper,
+  },
+}));
+
+const MinCollaps = ({
+  data,
+  isCollaps,
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/MinCollaps.tsx:109]:
```suggestion
  isCollapsed,
```

**Comment by bdauria** [comment]
@Noha-ElAbrouki I think we can get rid of the shadows and replace it with a slight border (same as the acutal menu)

**Comment by Noha-ElAbrouki** [comment]
> @Noha-ElAbrouki I think we can get rid of the shadows and replace it with a slight border (same as the acutal menu)

ook

**Comment by bdauria** [comment]
@Noha-ElAbrouki a few remarks:
- We should make the menu a bit more dense in order to optimize the spaced that is used. I think like it was previously (espacially in terms of width) was fine
- We also need to make sure the menu items are centered vertically
- There are a few animation glitches when hovering the items, as shown on the video. That might have something to do with memoization, but I'm not entirely sure. Can we try to make the elements appear without delay when hovering them?
- Same remark for the top logo when toggling the mode (collapsed vs plain)

Feb-28-2022 22-47-47

File: www/front_src/src/Navigation/Sidebar/Menu/MinCollaps.tsx [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+    backgroundColor: theme.palette.background.default,
+    left: ({ currentWidth }: StyleProps): string => theme.spacing(currentWidth),
+    minWidth: collapsWidth,
+    position: 'fixed',
+    top: ({ currentTop }: StyleProps): number | undefined => currentTop,
+    zIndex: theme.zIndex.mobileStepper,
+  },
+}));
+
+const MinCollaps = ({
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/MinCollaps.tsx:94]: MonCollaps doesn't sound clear to me, even though we're using the Collapse component under the hood. What about CollapsableItems?

File: www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+  const handleLeave = (): void => {
+    setHoveredIndex(null);
+    if (navigationItemSelected) {
+      Object.keys(navigationItemSelected).forEach(() => {
+        delete navigationItemSelected[`level_${level}`];
+      });
+    }
+  };
+
+  const isHover = (
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx:141]:
```suggestion
  const isItemHovered = (
```

File: www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+        object[levelTitle].index === index &&
+        object[levelTitle].label === item.label &&
+        object[levelTitle].url === item?.url
+      );
+    }
+
+    return false;
+  };
+
+  const getNestedIndex = (index, ind: number, content: Array<Page>): number => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx:158]: It's a bit confusing to have to parameters with almost the same names: index and ind. Can we find better names?

File: www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+      );
+    }
+    if (index === 1) {
+      return ind + Number(content[0].children?.length);
+    }
+
+    return ind;
+  };
+
+  const checkArray = (item: unknown): boolean => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx:173]:
```suggestion
  const isArrayItem = (item: unknown): boolean => {
```

File: www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+                    ? (): void => onClick(item)
+                    : undefined
+                }
+                onMouseEnter={(e: React.MouseEvent<HTMLElement>): void =>
+                  handleHover(e, index, item)
+                }
+              />
+            )}
+
+            {Array.isArray(item?.groups) && item.groups.length > 1 ? (
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx:257]:
```suggestion
            {isArrayItem(item?.groups) && item.groups.length > 1 ? (
```

File: www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+              <CollapsItem
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/CollapsItem.tsx:258]:
```suggestion
              <CollapsableItems
```

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 11]
```diff
@@ -41,33 +39,36 @@ const NavigationMenu = ({
   navigationData,
 }: Props): JSX.Element => {
   const classes = useStyles();
-  const theme = useTheme();
   const navigate = useNavigate();
 
   const [hoveredIndex, setHoveredIndex] = useState<number | null>(null);
   const [currentTop, setCurrentTop] = useState<number>();
+  const [maxHeightCollapsScroll, setMaxHeightCollapsScroll] = useState<
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:45]:
```suggestion
  const [collapseScrollMaxHeight, setCollapseScrollMaxHeight] = useState<
```

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 12]
```diff
@@ -41,33 +39,36 @@ const NavigationMenu = ({
...
+  const currentWidth = isDrawerOpen ? openedDrawerWidth / 8 : 8;
 
   const props = {
     currentTop,
     currentWidth,
     hoveredIndex,
     isDrawerOpen,
     level: 1,
+    maxHeightCollapsScroll,
+    setMaxHeightCollapsScroll,
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:65]:
```suggestion
    collapseScrollMaxHeight,
    setCollapseScrollMaxHeight,
```

**Pull request closed without merging**